code/modules/projectiles/guns/projectile.dm

```diff
@@ -180,7 +180,7 @@
 			silenced = A	//dodgy?
 			w_class = W_CLASS_MEDIUM
 			if(silencer_offset.len)
-				var/image/silence_overlay = image("icon" = 'icons/obj/gun_part.dmi', "icon_state" = "silencer_mounted")
+				var/image/silence_overlay = image("icon" = 'icons/obj/gun_part.dmi', "icon_state" = "[A.name]_mounted")
```

Comment:
Hand labeling a loudencer or silencer will break the icon_state and it will be invisible